### ISSUE TITLE
Advanced UI scaling please

### ISSUE THREAD

**Issue by Smedley-SC** [endless-sky/endless-sky#9048]
### Problem Description

On a high resolution screen, especially from far away, text (especially mission text or when inside the outfitter) can become very small, very fast.

especially with setups like 4K from 10-20 feet away

### Related Issue Links

I'm sure I'm not the first on all of these, I'll come and find as many related issues as possible, but right now my battery is low and I need to find my charger

searching for "scale" has turned up:

https://github.com/endless-sky/endless-sky/issues/3488

https://github.com/endless-sky/endless-sky/issues/1378

https://github.com/endless-sky/endless-sky/issues/805

### Desired Solution

0. prepack ES with compactlayout plugin, add it as a Display option
1. reorder "Display" and "Performance" settings so display is on bottom, to not be accidentally cut off if someone with a smaller screen tries setting it too high
2. additionally make hitting ESC on the menu screen redirect to display settings, for similar reasons, then cycle back to gameplay (unlikely) or reorder menu elements so that navigation to display settings remains near to the center of the screen
3. 300% or even 400% maximum for UI scale instead of 200% (potentially would even make 8K playable, though that's definitely not my monitor, nor any monitor I'm going to have anytime soon, maybe ever)
4. all text boxes scrollable, to make room for
5. resizeable text
6. custom font (import from file which font to prefer, revert to default in case a character is missing) (useful for people who would like to use fonts like Lexend or Dyslexie, maybe Atkinson Hyperlegible

### Alternative Approaches

move my monitor closer

use the mobile port (currently am)

### Additional Context

Take a screenshot of ES at 100% scaling, full screen
now add a massive bottom and right border so that it only takes up one quadrant of the new image (to simulate text not scaling with resolution, mine is set at 200% at 4K, which would equate to 100% at 1920*1080)
Now add new borders left and right, so that the previous image only takes up 1/3rd of your new image horizontally  (to simulate distance from monitor)
full screen to fit, no stretch
current situation here

**Comment by lumbar527** [comment]
> 2. additionally make hitting ESC on the menu screen redirect to display settings, for similar reasons, then cycle back to gameplay (unlikely) or reorder menu elements so that navigation to display settings remains near to the center of the screen

I think that would cause other problems. When I hit Escape, it is either to 1) Reload my save, 2) Change weapon settings (auto-fire and stuff).

**Comment by Smedley-SC** [comment]
yeah, very unlikely indeed

I just think a "let the player out of stupid decision" approach is a little better than the "prevent player from making stupid decision" approach we have now
especially considering other aspects of the game it affects, and some of the limits it imposes, which are outside of the scope of this feature request

**Comment by Zitchas** [comment]
Yes, Endless Sky doesn't handle scaling and zooming very well. There's been a few other attempts at improving some aspects of those, and those have turned up *other* oddities.

**Comment by Hurleveur** [comment]
Well at least the map planet display adapts well enough, right?
One could probably do a lot of improvements in a plugin by modifying interface.txt?

**Comment by Amazinite** [comment]
> reorder "Display" and "Performance" settings so display is on bottom, to not be accidentally cut off if someone with a smaller screen tries setting it too high

The game already limits you from zooming in so far as to cause the preferences to overflow the screen.

> 300% or even 400% maximum for UI scale instead of 200% (potentially would even make 8K playable, though that's definitely not my monitor, nor any monitor I'm going to have anytime soon, maybe ever)

Resolved by #12393.

>all text boxes scrollable, to make room for resizeable text

Will be resolved by #12446.

I'll also be adding a way to increase the font size used for most areas of the game. #12448.
